omegaconf/_utils.py

```diff
@@ -330,22 +333,25 @@ def get_value_kind(value: Any) -> ValueKind:
                    "ftp://${host}/path", "${foo:${bar}, [true], {'baz': ${baz}}}"
 
     :param value: Input to classify.
+    :param strict_interpolation_validation: If `True`, then when `value` is a string
+        containing "${", it is parsed to validate the interpolation syntax. If `False`,
+        this parsing step is skipped: this is more efficient, but will not detect errors.
     """
 
     value = _get_value(value)
 
     if value == "???":
         return ValueKind.MANDATORY_MISSING
 
-    # We detect potential interpolations simply by the presence of "${" in the string.
-    # This is much more efficient than calling `grammar_parser.parse()`, but one must
-    # be aware that:
-    #   (a) invalid interpolations are not detected here
-    #   (b) escaped interpolations (ex: "esc: \${bar}") are identified as interpolations
-    # These are not actually problems though, since as soon as "${" is present in a
-    # string, it *has* to be parsed as an interpolation to properly (a) detect errors,
-    # and (b) un-escape any escaped interpolations.
+    # We identify potential interpolations by the presence of "${" in the string.
+    # Note that escaped interpolations (ex: "esc: \${bar}") are identified as
+    # interpolations: this is intended, since they must be processed as interpolations
+    # for the string to be properly un-escaped.
+    # Keep in mind that invalid interpolations will only be detected when
+    # `strict_interpolation_validation` is True.
     if isinstance(value, str) and "${" in value:
+        if strict_interpolation_validation:
+            parse(value)  # will raise an error if syntax is invalid
```

Comment:
doesn't feel right that an is_function can raise an exception.
Shouldn't the type of invalid interpolations be ValueKind.VALUE?

Reply:
As discussed, an invalid interpolation is not a valid `ValueKind.VALUE`. So it actually makes sense to raise an exception here (or, alternatively, we could add a new `ValueKind.INVALID`)